### PR TITLE
Fixed #26093 -- Allowed escape sequences extraction by gettext on Python 3

### PR DESCRIPTION
**Pull request by claudep**
Thanks Sylvain Fankhauser for the report.

### CODE REVIEW

File: django/utils/translation/trans_real.py [hunk 1]
```diff
@@ -566,6 +566,8 @@ def templatize(src, origin=None):
     comment = []
     lineno_comment_map = {}
     comment_lineno_cache = None
+    # Adding the u prefix on Python 3 allows gettext to recognize the Unicode string (#26093).
```
Comment (timgraham) [inline django/utils/translation/trans_real.py]: Maybe we could omit "on Python 3" so we don't have to edit the comment when dropping Python 2 support. Also wrap comment at 79 characters.

File: django/utils/translation/trans_real.py [hunk 2]
```diff
@@ -597,26 +599,30 @@ def join_tokens(tokens, trim=False):
                 if endbmatch:
                     if inplural:
                         if message_context:
-                            out.write(' npgettext(%r, %r, %r,count) ' % (
+                            out.write(' npgettext({p}{!r}, {p}{!r}, {p}{!r},count) '.format(
                                 message_context,
                                 join_tokens(singular, trimmed),
-                                join_tokens(plural, trimmed)))
+                                join_tokens(plural, trimmed),
+                                p=raw_prefix))
```
Comment (timgraham) [inline django/utils/translation/trans_real.py]: care to put the closing parentheses on the next line to follow style we use elsewhere?

File: django/utils/translation/trans_real.py (hunk 1 above)
Comment (claudep) [inline django/utils/translation/trans_real.py]: Fair. Are comments subject to the 79-chars hard limit?

Comment (timgraham) [inline django/utils/translation/trans_real.py]: I've been doing that for docstrings and comments.

**Comment by last-partizan** [comment]
Technically, it was bug in gettext and it is fixed now (but not released yet, probably will be included in the next version after 0.22.5).

https://savannah.gnu.org/bugs/?65778 

Somewhere in the future when everyone uses new gettext this workaround may no longer be needed.

**Comment by claudep** [comment]
Thanks, great to know! It might be worth completing the comment in the source code, as I fear your comment here might get forgotten or even lost. What do you think?

**Comment by timgraham** [comment]
Here's a reminder: https://code.djangoproject.com/ticket/35684

**Pull request merged**